rero_ils/modules/documents/views.py

```diff
@@ -342,17 +327,27 @@ def item_library_pickup_locations(item):
     """Get the pickup locations of the library of the given item."""
     location_pid = item.replace_refs()['location']['pid']
     location = Location.get_record_by_pid(location_pid)
-    library_pid = location.replace_refs()['library']['pid']
-    library = Library.get_record_by_pid(library_pid).replace_refs()
-    organisation = Organisation.get_record_by_pid(
-        library['organisation']['pid'])
-    locations = []
-    for library in organisation.get_libraries():
-        location = Location.get_record_by_pid(
-            library.get_pickup_location_pid())
-        if location:
-            locations.append(location)
-    return locations
+    # Either the location defines some 'restrict_pickup_to' either not.
+    # * If 'restrict_pickup_to' is defined, then only these locations are
+    #   eligible as possible pickup_locations
+    # * Otherwise, get all organisation pickup locations of the item belongs to
+    if 'restrict_pickup_to' in location:
+        location = location.replace_refs()
+        # Get all pickup locations as Location objects and append it to the
+        # location item (removing possible None values)
+        pickup_locations = list(filter(None, [
```

Comment:
can be replaced by or can make sure of `location.restrict_pickup_to?`

Reply:
done